src/credentials/auth.ts

```diff
@@ -586,6 +587,48 @@ export class Auth implements AuthService, ConnectionManager {
         return this.#validationErrors.get(connection.id)
     }
 
+    /**
+     * Returns true if credential data can be authenticated successfully
+     */
+    public async canAuthenticateRawData(data: StaticProfile): Promise<boolean> {
+        const tempId = await this.addTempCredential(data)
+        const tempIdString = asString(tempId)
+        try {
+            await this.reauthenticate({ id: tempIdString })
+        } catch (e) {
+            if (e instanceof Error && e.name === 'InvalidClientTokenId') {
```

Comment:
is this checking a ToolkitError or is the `name` coming from the service? is there a `instanceof` comparison we can use instead?

Reply:
This is an error from the sdk, but I cannot find an error object for `InvalidClientTokenId` in the sdk types.

I've updated to use `isAwsError()` instead and then check the `code` instead of `name`